File: Domain/Gallery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Gallery
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public ICollection<Photo> Photos { get; set; }
    }
}

File: client-app/src/app/models/gallery.ts
import { Photo } from "./photo";

export interface Gallery {
    id: string;
    name: string;
    description: string;
    photos?: Photo[];
}

export class Gallery {
    id = "";
    name = "";
    description = "";
    photos?: Photo[] = [];
    constructor (gallery?: Gallery) {
        Object.assign(this, gallery);
    }
}